Fix examples in Control flow effects of await

Worldwidebrine opened the pull request:
1. Fixes examples in Control flow effects of await.
2. Squashes commit https://github.com/Worldwidebrine/content/commit/9a771cdf71361a2d7528fa729e1512461a97db28 and commit https://github.com/Worldwidebrine/content/commit/12089e4fd80bebc58c30a3c70a5945d49cf3d205, and fixes typo in the commit message.

check not passed. okay

Josh-Cena commented:
@Worldwidebrine if you need help creating the PR and/or writing the content just let me know

Worldwidebrine commented:
Sorry but I didn't want to spam. I was about to give up fixing commit message typo so I deleted my commits.
But after I dropped my commits, github closed my draft. https://github.com/orgs/community/discussions/38057
`Control flow effects of await` is full of wrong examples.
@Josh-Cena 
Correction for example 2:
code before the first `await` is sync, not in `then()`.
(ignore error, more readable)
```js
function foo(name) {
  console.log(name, "start");
  console.log(name, "middle");
  console.log(name, "end");
  return Promise.resolve();
}
```
Correction for example 4:
this too. code before the first `await` is sync.
(let constructor catch error)
```js
function foo(name) {
  return new Promise((resolve) => {
      console.log(name, "start");
      resolve(console.log(name, "middle"));
    })
    .then(() => {
      console.log(name, "end");
    });
}
```
And then, this example needs to be added:
because of the previous corrections, there is now no example for
> While the extra `then()` handler is not necessary and can be merged with the previous one, its existence means the code will take one extra tick to complete. The same happens for `await`.

```js
async function foo(name) {
  console.log(name, "start");
  await null;
  console.log(name, "middle");
  await null;
  console.log(name, "end");
}

foo("First");
foo("Second");

// First start
// Second start
// First middle
// Second middle
// First end
// Second end
```
corresponds to
```js
function foo(name) {
  return new Promise((resolve) => {
      console.log(name, "start");
      resolve(null);
    })
    .then(() => {
      console.log(name, "middle");
      return null;
    })
    .then(() => {
      console.log(name, "end");
    });
}
```
And add example for
> Therefore, make sure to use `await` only when necessary (to unwrap promises into their values).

other microtasks can execute before async function resumes:
```js
let i = 0;
queueMicrotask(function test() {
  i++;
  console.log("microtask", i);
  if (i < 5) {
    queueMicrotask(test);
  }
});
(async () => {
  console.log("start");
  await null;
  console.log("resume", 1);
  for (let i = 2; i < 5; i++) {
    await null;
    console.log("resume", i);
  }
  await null;
  console.log("end");
})();
queueMicrotask(() => {
  console.log("queueMicrotask() after calling async function");
})
console.log("sync");
```

Josh-Cena commented:
Thanks. Do you want me to send a PR for you?

Worldwidebrine commented:
It would be greate if you can make it.
I'm so confused by git.
